### ISSUE TITLE
setup: unplug-to-identify cannot see a camera whose color node lost the udev by-id name race

### ISSUE THREAD

**Issue by jeqcho** [robocurve/inspect-robots#261]
## Symptom

On a rig with a RealSense D435 as the top camera, `inspect-robots setup` shows the saved by-id path as `(current, not detected)`, lists only 3 of 4 cameras, and `u` (unplug to identify) always reports:

```
no camera device disappeared; unplug one camera and try again
```

while `dmesg` records every disconnect/reconnect cleanly. Replugging and swapping cables does not help. The camera is enumerated and healthy the whole time (`lsusb`: `8086:0b07` at SuperSpeed).

## Root cause

1. The D435 exposes depth and RGB as two separate UVC interfaces on one USB device. udev's `60-persistent-v4l.rules` builds by-id names from the per-interface `index` attribute, so both interfaces race for `...-video-index0/1` and one silently overwrites the other. Which interface wins re-rolls on every plug event. On the affected rig the RGB capture node (`/dev/video10`, the device's only color-capable node) currently has no by-id symlink at all; the surviving by-id links point at depth (`Z16`), IR, and metadata nodes.
2. `_scan_cameras` (`src/inspect_robots/_setup.py:793`) filters the listing to color-capable nodes via `_v4l2_color_capture`. With the RGB node's symlink gone, the D435 contributes zero rows to the by-id listing, so it cannot be selected and the saved config value reads as not detected.
3. `_identify_by_replug` (`src/inspect_robots/_setup.py:291`) diffs the pre-filtered listing of the active directory. A camera absent from the "before" list can never disappear, so `u` fails with "no camera device disappeared" even though the kernel saw the unplug.
4. Switching to the by-path listing with `p` does not rescue `u`: newer systemd creates `usbv2-`/`usbv3-` alias symlinks next to the plain `usb-` names, both resolving to the same node. One physical unplug removes 2+ rows and trips the `len(disappeared) > 1` guard (`src/inspect_robots/_setup.py:298`), telling the user to "unplug only one".

D405s never hit this because they expose all six video nodes through a single UVC interface (unique `index` attrs, no name collision). Any multi-interface UVC camera (D435/D455, likely others) is affected.

A related case on the same rig: two Innomaker U20CAMs ship the identical serial `SN0001`, so only one gets by-id links and the other is reachable by-path only.

## Verified on hardware

- `lsusb -t`: D435 on bus 4 port 9 at 5000M, all 5 interfaces bound to uvcvideo.
- Kernel log: three unplug/replug cycles, each with clean re-enumeration, while the wizard reported no change.
- The wizard's own probe (`_v4l2_color_capture`) confirms `/dev/video10` is the only color node (`YUYV`); `video0`/`video2` are `Z16`/`GREY`.
- `/dev/v4l/by-id` for the D435 after the last replug: `index0 -> video0` (depth), `index1 -> video11` (metadata), no link to `video10`.

## Proposed fix

Diff physical USB devices, not symlink listings; use symlinks only to choose the stored name.

1. Scan `/sys/class/video4linux/*` and group video nodes by parent USB device (port path + vendor/product/serial). Probe color capability on the raw `/dev/videoN` nodes, then attach whichever by-id/by-path symlinks resolve to them. A camera with no by-id link still becomes a full record.
2. `u` diffs the set of USB devices that had color-capable nodes. Alias duplication becomes irrelevant and a camera invisible in by-id is still identifiable. Only devices with video nodes count toward the "unplug only one" guard.
3. Choose the stored path after replug (the by-id race re-rolls, so decide late), with a trust ladder: a by-id symlink that resolves to one of this device's color nodes and whose serial is unique on the bus; else the plain `usb-` by-path name of a color node; else the raw node with a warning. The shared-serial Innomaker pair fails the uniqueness check and lands on by-path deterministically.
4. List one row per physical camera (best trusted name), so a camera whose by-id links are broken still shows up, and reconcile a saved-but-missing config path against scanned serials instead of reporting only "not detected".
5. Poll briefly for udev symlink settle after replug before choosing the name.

Non-Linux and no-sysfs environments keep the current directory-listing behavior (records keyed by resolved target, including the existing list-everything fallback when no color node is confirmed).
